### DISCUSSION TITLE
Displaying time series from GeoJSON

### DISCUSSION THREAD

**Discussion by hakoncarlsen** [TerriaJS/terriajs#5798] [Q&A] [answered]
Hi! Is it possible to choose display variables with GeoJSON files in the same way as with csv files?

Clarification:

I have the following GeoJSON file, which I wish to display:

```JSON
{
    "type": "FeatureCollection",
    "features": [
        {
            "geometry": {
                "type": "Point",
                "coordinates": [
                    5.734,
                    58.969
                ]
            },
            "properties": {
                "time": "2021-09-07T13:00:00Z",
                "temperature": 17.9,
                "wind speed": 3.7
            },
            "type": "Feature"
        },
        {
            "geometry": {
                "type": "Point",
                "coordinates": [
                    5.734,
                    58.969
                ]
            },
            "properties": {
                "time": "2021-09-07T14:00:00Z",
                "temperature": 17.6,
                "wind speed": 4.0
            },
            "type": "Feature"
        }
    ]
}
```

I use the following catalog item in the initializaion file:

```JSON

{
    "type": "geojson",
    "name": "my_weather_data geojson",
    "url": "test/my_weather_data.geojson",
    "timeProperty": "time"
}
```

I also have a csv file that contains the same data as the GeoJSON file:

| time                 | temperature | wind speed | latitude | longitude |
|----------------------|-------------|------------|----------|-----------|
| 2021-09-07T13:00:00Z | 17.9        | 3.7        | 58.969   | 5.734     |
| 2021-09-07T14:00:00Z | 17.6        | 4.0        | 58.969   | 5.734     |
| 2021-09-07T15:00:00Z | 17.6        | 3.4        | 58.969   | 5.734     |

I display this csv file by using the following catalog item in the initialization file:

```JSON
{
      "type": "csv",
      "name": "my_weather_data csv",
      "url": "test/my_weather_data.csv"
}
```

When I import both of these data sets into the TerriaMap client, this is the result:

![csv_vs_geojson_weather_data](https://user-images.githubusercontent.com/89078335/132354485-23b7f7c9-98dc-4051-a220-c58d7d194557.png)

I am able to see the differences in temperature or wind speed over time by using the csv file. But I am not when using the GeoJSON file.  The only thing TerriaJS displays from the GeoJSON file is the location of the data points, and not the properties. 

Also, what if the GeoJSON file is on the following format instead:

```JSON
{

	"type": "Feature",
	"geometry": {
		"type": "Point",
		"coordinates": [5.734, 58.969]
	},
	"properties": {
		"timeseries": [{
			"time": "2021-09-07T13:00:00Z",
			"data": {
				"instant": {
					"details": {
						"temperature": 17.9,
						"wind_speed": 3.7
					}
				}
			}
		}, {
			"time": " 2021-09-07T14:00:00Z",
			"data": {
				"instant": {
					"details": {
						"temperature": 17.6,
						"wind_speed": 4.0
					}
				}
			}
		}]
	}
}
```

Is it somehow possible to specify in the initialization file, where TerriaJS needs to find the time, temperature and wind speed data, and to display it in a similar manner to the csv file?

**Comment by nf-s** [comment, marked as answer]
Hi @hakoncarlsen 

Currently we don't have ability to visualise GeoJSON feature properties in the same way as CSV columns.
But it is in the works! 
We plan to introduce this feature in the coming months

Thanks,
Nick

**Reply by hakoncarlsen** [reply]
Ok, I understand. 
Do you know if it possible to display the feature properties of GeoJSON over time on the map at all, even if you cannot do it in the same way as with a csv?

Thanks a lot for you answer!
Håkon

**Reply by nf-s** [reply]
Hi @hakoncarlsen 

Sorry for the delayed reply!
At the moment, the only way would be to specify style manually in GeoJSON per feature using the [simple style spec](https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0) - this can be combined with `timeProperty`.

For example - http://ci.terria.io/main/#share=s-5ewNQj86ckl8Cjep2lye8SJGNjP

```json
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {
        "stroke": "#555555",
        "stroke-width": 2,
        "stroke-opacity": 1,
        "fill": "#aa0084",
        "fill-opacity": 0.5
      },
      "geometry": {
        "type": "Polygon",
        "coordinates": [
          [
            [
              130.60546875,
              -16.97274101999901
            ],
            [
              123.3984375,
              -29.382175075145277
            ],
            [
              149.765625,
              -32.69486597787506
            ],
            [
              145.37109375,
              -13.923403897723334
            ],
            [
              130.60546875,
              -16.97274101999901
            ]
          ]
        ]
      }
    }
  ]
}
```

I realise this is far from ideal, hopefully we have proper styling soon!

**Reply by hakoncarlsen** [reply]
Allright, thanks for your response!

**Reply by LHKaay** [reply]
@nf-s Hi,
I can not find any document tell about this feature. So, Are there any update on this useful feature ?

**Reply by uwer** [reply]
same here - are there any updates ? documentation ?
